### ISSUE TITLE
Python-3.8.2-GCCcore-9.3.0.eb does not build (on CentOS 8.2)

### ISSUE THREAD

**Issue by patoddh** [easybuilders/easybuild-easyconfigs#11093]
Build fails with "[Errno 11] Resource temporarily unavailable". Here's the end of the stdout:

== taking care of extensions...
== installing extension pip 20.0.2 (1/77)...
== installing extension setuptools 45.2.0 (2/77)...
== FAILED: Installation ended unsuccessfully (build directory: /grid/it/data/elzar/easybuild/build/Python/3.8.2/GCCcore-9.3.0): build failed (first 300 chars): cmd " pip install --prefix=/grid/it/data/elzar/easybuild/software/Python/3.8.2-GCCcore-9.3.0  --no-deps  --ignore-installed  --no-build-isolation  ." exited with exit code 1 and output:
Processing /grid/it/data/elzar/easybuild/build/Python/3.8.2/GCCcore-9.3.0/setuptools/setuptools-45.2.0
ERROR: Coul (took 6 min 44 sec)
== Results of the build can be found in the log file(s) /tmp/eb-bm0by3sc/easybuild-Python-3.8.2-20200811.113957.fadLg.log
ERROR: Build of /grid/it/data/elzar/easybuild/software/EasyBuild/4.2.2/easybuild/easyconfigs/p/Python/Python-3.8.2-GCCcore-9.3.0.eb failed (err: 'build failed (first 300 chars): cmd " pip install --prefix=/grid/it/data/elzar/easybuild/software/Python/3.8.2-GCCcore-9.3.0  --no-deps  --ignore-installed  --no-build-isolation  ." exited with exit code 1 and output:\nProcessing /grid/it/data/elzar/easybuild/build/Python/3.8.2/GCCcore-9.3.0/setuptools/setuptools-45.2.0\nERROR: Coul')'

The log file is attached below. If I remove the setuptools extension, the same error occurs later for the distlib extension.

Interestingly, the older Python-3.7.4-GCCcore-8.3.0.eb DOES build OK for me. I tried using its the setuptools version 41.2.0 for this newer Python but got the same error.

Log file:

[easybuild-Python-3.8.2-20200811.113957.fadLg.log](https://github.com/easybuilders/easybuild-easyconfigs/files/5057980/easybuild-Python-3.8.2-20200811.113957.fadLg.log)

**Comment by patoddh** [comment]
The error message is misleading, see: https://bugs.python.org/issue25473

I have seen this also in PGO regression testing when optimization is on when building Python with GCCcore 8.3.0 (had to set optimization=False there). But this Python build with GCCcore 9.3.0 got by that OK (default optimization True), only to hit this problem on the extensions. Go figure.

**Comment by boegel** [comment]
I haven't seen problems with installing Python 3.8.2 with `GCCcore/9.3.0` on CentOS 8.2, but maybe this is related to the problems with `GCCcore` after upgrading to CentOS 8.2, just like in #11032 (more details available in #10666)?

It could be working with `GCCcore/8.3.0` because you've reinstalled that already, as you reported in #11032?

**Comment by patoddh** [comment]
I think it is a different issue than #11032. I only just built GCCcore/9.3.0 and foss-2020a last week for the first time, on CentOS 8.2.

The weird thing is that this Python 3.8.2 built with optimization on (until the error when doing extensions), while I had to turn off optimization in order to get Python 3.7.4 to build with GCCcore 8.3.0. In that case the build hung with 2 PGO regression test processes, with strace showing one of them in an infinite loop of error 11, resource temporarily unavailable). And the extensions built fine in that case with optimization set to False to work around the PGO test thing.

And in this case of Python 3.8.2, the error when building the extension is also error 11, resource temporarily unavailable.

**Comment by boegel** [comment]
@patoddh So does disabling the optimizations fix the problems you're seeing with Python 3.8.2 then?

I wonder if we can somehow detect that the optimizations should not be enabled?

Also, I'm puzzled why you are reporting this while it's working fine for many others...

**Comment by patoddh** [comment]
@boegel No, disabling the optimizations does not fix the problem with Python 3.8.2 (GCCcore 9.3.0). It only fixed the issue with Python 3.7.4 (GCCcore 8.3.0), which was hanging in the "building" stage. In contrast Python 3.8.2 gets by the "building" stage with optimization on. Then Python 3.8.2 errors out on the setuptools extension (regardless of the optimization).

I'm puzzled too. Only idea would be that I'm using GPFS.

**Comment by patoddh** [comment]
This problem also occurs building SciPy-bundle-2020.03-foss-2020a-Python-3.8.2.eb (on the scipy extension).

**Comment by boegel** [comment]
@patoddh I checked this again, and I think the problem may be that you are using a shared filesystem for the build directories... I see that EasyBuild is using `/grid/it/data/elzar/easybuild/build`, which doesn't look like a local filesystem to me.

Can you share the output of `eb --show-config`?

And can you try using a local filesystem like `/tmp` or ramdisk (`/dev/shm`) for the build directories?

**Comment by patoddh** [comment]
@boegel I had changed it from a shared filesystem to /dev/shm a few months back:

buildpath                 (F) = /dev/shm

Since I don't remember what I tested after making the change, and I already have Python-3.8.2-GCCcore-9.3.0.eb (title of this issue) built via pip workarounds, I just tried building SciPy-bundle-2020.03-foss-2020a-Python-3.8.2.eb again.

USing EasyBuild 4.3.3, I'm getting numpy test failures now. According to a 12/15/20 thread on the email list about SciPy-bundle for the Intel toolchain, I tried "ignore_test_result = True" in SciPy-bundle-2020.03-foss-2020a-Python-3.8.2.eb, but Easybuild complained aboput an unknown easyconfig parameter.
